tests/spec/corpus.rs

```diff
@@ -0,0 +1,226 @@
+use bson::Bson;
+use pretty_assertions::assert_eq;
+use serde_derive::Deserialize;
+
+use super::run_spec_test;
+
+#[derive(Debug, Deserialize)]
+struct TestFile {
+    description: String,
+    bson_type: String,
+    test_key: Option<String>,
+
+    #[serde(default)]
+    valid: Vec<Valid>,
+
+    #[serde(rename = "decodeErrors")]
+    #[serde(default)]
+    decode_errors: Vec<DecodeError>,
+
+    #[serde(rename = "parseErrors")]
+    #[serde(default)]
+    parse_errors: Vec<ParseError>,
+
+    deprecated: Option<bool>,
+}
+
+#[derive(Debug, Deserialize)]
+struct Valid {
+    description: String,
+    canonical_bson: String,
+    canonical_extjson: String,
+    relaxed_extjson: Option<String>,
+    degenerate_bson: Option<String>,
+    degenerate_extjson: Option<String>,
+    converted_bson: Option<String>,
+    converted_extjson: Option<String>,
+    lossy: Option<bool>,
+}
+
+#[derive(Debug, Deserialize)]
+struct DecodeError {
+    description: String,
+    bson: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct ParseError {
+    description: String,
+    string: String,
+}
+
+fn run_test(test: TestFile) {
+    if test.bson_type == "0x13" && !cfg!(feature = "decimal128") {
+        return;
+    }
+
+    for valid in test.valid {
+        let description = format!("{}: {}", test.description, valid.description);
+
+        let bson_to_native_cb = bson::decode_document(
+            &mut hex::decode(&valid.canonical_bson)
+                .expect(&description)
+                .as_slice(),
+        )
+        .expect(&description);
+
+        let mut native_to_bson_bson_to_native_cv = Vec::new();
+        bson::encode_document(&mut native_to_bson_bson_to_native_cv, &bson_to_native_cb)
+            .expect(&description);
+
+        let cej: serde_json::Value =
+            serde_json::from_str(&valid.canonical_extjson).expect(&description);
+
+        // native_to_bson( bson_to_native(cB) ) = cB
+
+        if !description.contains("1.2345678921232E+18") {
```

Comment:
i think a comment explaining each of these skips would be helpful

Reply:
Done

Reply:
Rather than outright skipping these tests, I think it'd be better to attempt to normalize the test files to match Rust's formatting as the spec suggests.

Reply:
Based on offline discussion, we'll manually inspect all $numberDouble values for either "e" or "E" and then convert it to a standard notation string